src/azure-cli-core/azure/cli/core/keys.py

```diff
@@ -32,14 +35,34 @@ def is_valid_ssh_rsa_public_key(openssh_pubkey):
 def generate_ssh_keys(private_key_filepath, public_key_filepath):
     import paramiko
 
+    try:
+        with open(public_key_filepath, 'r') as public_key_file:
+            public_key = public_key_file.read()
+            pub_ssh_dir, _ = os.path.split(public_key_filepath)
```

Comment:
Since the `basename` is not used here just use `os.path.dirname`

https://docs.python.org/3/library/os.path.html?highlight=os%20path%20dirname#os.path.dirname